### PR TITLE
rethrown critical ValueError, AttributeError and TypeError

### PR DESCRIPTION
**Pull request by guotuofeng**
## Describe your changes
Normalize ValueError exception handling. Most of time, user errors need not be caught. So, the engine run need explicitly ignore it.

## Checklist before requesting a review
- [x] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [x] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

**Comment by xiaoyu-work** [comment]
There are other errors that may need to be aware: like [`ImportError`](https://aiinfra.visualstudio.com/PublicPackages/_build/results?buildId=317839&view=logs&j=dbbdd4d8-ffd3-515a-31fd-659375dbdff8&t=13dad60e-7187-5e88-dbc0-776ed0850a3f&l=787) and `AttributeError` or maybe more. All these errors may indicate the code is wrong. Are you planning to add other errors here?

**Comment by guotuofeng** [comment]
> There are other errors that may need to be aware: like [`ImportError`](https://aiinfra.visualstudio.com/PublicPackages/_build/results?buildId=317839&view=logs&j=dbbdd4d8-ffd3-515a-31fd-659375dbdff8&t=13dad60e-7187-5e88-dbc0-776ed0850a3f&l=787), [`TypeError`](https://aiinfra.visualstudio.com/PublicPackages/_build/results?buildId=317839&view=logs&j=dbbdd4d8-ffd3-515a-31fd-659375dbdff8&t=13dad60e-7187-5e88-dbc0-776ed0850a3f&l=809), `AttributeError` or maybe more. All these errors may indicate the code is wrong. Are you planning to add other errors here?

just add these errors. For other errors, we can add them when we have the knowledge in future. Basically, exception handling is a keep-on-going work.

**Review by jambayk — APPROVED** [review]

File: olive/engine/engine.py [hunk 1]
```diff
@@ -861,6 +863,9 @@ def _run_pass(
                 input_model = self._prepare_non_local_model(input_model)
             try:
                 output_model = host.run_pass(p, input_model, output_model_path, pass_search_point)
+            except (ValueError, TypeError, AttributeError):
+                # Don't catch these errors since most of time, it is caused by the user errors and need not retry.
+                raise
             except Exception:
                 output_model = PRUNED_CONFIG
                 # TODO: from the time being, we need to catch all exceptions to make the
```
Comment (trajepl) [inline olive/engine/engine.py:871]: Nit: Shall we remove this comments? As you already fix it.

Comment (guotuofeng) [reply olive/engine/engine.py:871]: actually, it is still need some works to make sure the exception handling fully done. So I keep the TODO here

**Pull request merged**